packages/modules/order/src/services/__tests__/util/actions/credit-line-add.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { ChangeActionType } from "@medusajs/framework/utils"
+import { VirtualOrder } from "@types"
+import { calculateOrderChange } from "../../../../utils"
+
+describe("Action: Credit Line Add", function () {
+  const originalOrder: VirtualOrder = {
+    id: "order_1",
+    items: [
+      {
+        id: "item_1",
+        quantity: 1,
+        unit_price: 10,
+        compare_at_unit_price: null,
+        order_id: "1",
+
+        detail: {
+          quantity: 1,
+          order_id: "1",
+          delivered_quantity: 1,
+          shipped_quantity: 1,
+          fulfilled_quantity: 1,
+          return_requested_quantity: 0,
+          return_received_quantity: 0,
+          return_dismissed_quantity: 0,
+          written_off_quantity: 0,
+        },
+      },
+    ],
+    shipping_methods: [
+      {
+        id: "shipping_1",
+        amount: 20,
+        order_id: "1",
+      },
+    ],
+    credit_lines: [],
+    total: 30,
+  }
+
+  /*
+    We have an original order with a total of 30, the summary would then be the following:
+
+    {
+      "transaction_total": 0,
+      "original_order_total": 30,
+      "current_order_total": 30,
+      "pending_difference": 30,
+      "difference_sum": 0,
+      "paid_total": 0,
+      "refunded_total": 0,
+      "credit_line_total": 0
+    }
+
+    Upon adding a credit line, the order total and the pending difference will increase making it possible for the merchant
+    to request the customer for a payment for an arbitrary reason, or prepare the order balance sheet to then allow
```

Comment:
question: I wouldn't have expected credit lines to be used for additional payments, only refunds–can you expand on this?

Reply:
That sounds right yeah

Reply:
Do you want to update this branch or do it in a follow up?

Reply:
Can do either. Making good progress rn, so focusing on not hindering that flow. 😄